### DISCUSSION TITLE
HTTP POST image, how to interact with it?

### DISCUSSION THREAD

**Discussion by darrenleeleelee1** [fastapi/fastapi#7169] [Questions] [answered]
### Example
```Python
from fastapi import FastAPI, File, UploadFile
from fastapi.responses import FileResponse
from pydantic import BaseModel
import numpy as np
import shutil
app = FastAPI()


@app.get("/")
async def main():
    return FileResponse(r"misdo.jpg")


@app.post("/uploadfile/")
async def create_upload_file(image: UploadFile = File(...)):
    with open("destination.jpg", "wb") as buffer:
         shutil.copyfileobj(image.file, buffer)
    return image.filename`
```

### Description

* This code is work fine in 127.0.0.1:8000/docs.
* But I am wondering how can I post without using 127.0.0.1:8000/docs.
* More specified is how could I use other languague(Swift) to interact with this API.
* I know how to do with HTTP.GET, but HTTP.POST still bothering me.

* Or is my POST funtion is not well defined?
### Environment

* macOS
* FastAPI Version: 0.61.1
* Python version: 3.8.6

**Comment by ycd** [comment, marked as answer]
> But I am wondering how can I post without using 127.0.0.1:8000/docs.

You can use cURL.

```bash
curl -X POST "http://127.0.0.1:8000/uploadfile/" -H  "Content-Type: multipart/form-data" -F "file=@your_file_name;type=image/jpeg"
```

I would use aiofiles with `async with` in `/uploadfile`.

```python
import aiofiles

@app.post("/uploadfile/")
async def cache_favicon(file: UploadFile = File(...)):
    img = await file.read()
   
    async with aiofiles.open("destination.jpg" , "wb") as f:
        await f.write(img)
```

**Comment by Gokulnath31** [comment]
How to convert the uploaded image  to Numpy array?

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! :clap: :bow: 

Thanks for reporting back and closing the issue @darrenleeleelee1  :+1:

@Gokulnath31 you can probably search the issues and find an answer: https://github.com/tiangolo/fastapi/issues?q=is%3Aissue+numpy+image+is%3Aclosed+ but if you still have problems, please create a new issue following the template.
